channels/encomsp/client/encomsp_main.c

```diff
@@ -980,7 +1003,5 @@ BOOL VCAPITYPE VirtualChannelEntry(PCHANNEL_ENTRY_POINTS pEntryPoints)
 	encomsp->channelEntryPoints.pInterface = *(encomsp->channelEntryPoints.ppInterface);
 	encomsp->channelEntryPoints.ppInterface = &(encomsp->channelEntryPoints.pInterface);
 
-	encomsp_add_init_handle_data(encomsp->InitHandle, (void*) encomsp);
-
-	return 1;
+	return encomsp_add_init_handle_data(encomsp->InitHandle, (void*) encomsp) ? 1 : -1;
```

Comment:
This function (VirtualChannelEntry) is BOOL. Just return encomsp_add_init_handle_data() - right ?